### ISSUE TITLE
Headed mode inside Docker Container

### ISSUE THREAD

**Issue by itsperini** [QIN2DIM/undetected-playwright#13]
Hey,
I have installed the library inside [this](https://hub.docker.com/_/microsoft-playwright-python/) docker container and when I try to run playwright in headed mode I get the following error:

  ```bash
  Looks like you launched a headed browser without having a XServer running.
  Set either 'headless: true' or use 'xvfb-run <your-playwright-app>' before running Playwright.
  ```

I know the container is not connected to any display, is there a workaround with this library? Thanks

Current OS: Windows 11

**Comment by alternativshik** [comment]
add xserver to docker container or or use 'xvfb-run <your-playwright-app>'

**Comment by gexiea** [comment]
Yeah xvfb-run works to bypass the error, but if you actually need to see the headed browser to debug captchas or whatever, you'll need a VNC server anyway. I got tired of configuring X11 and noVNC manually in Docker, so I just pull the base image from stealthbrowser.org now.

It has the desktop environment and noVNC pre-configured, so you just map a port and watch the headed browser run in your own browser. The nice side effect is their custom Chromium is patched at the source level, so it naturally drops a lot of the standard bot characteristics that trigger captchas in the first place.

They actually charge per fingerprint profile if you use their full service, but I'm cheap, so I just spin up their base container and use Playwright/JS to spoof the fingerprints myself. It's obviously not as thorough as their native bottom-level spoofing, but combined with their patched browser base, it passes most checks and completely solves the Docker UI headache.
